docs/sdk/src/reference_docs/development_environment_advice.rs

```diff
@@ -111,3 +113,38 @@
 //! If you have a powerful remote server available, you may consider using
 //! [cargo-remote](https://github.com/sgeisler/cargo-remote) to execute cargo commands on it,
 //! freeing up local resources for other tasks like `rust-analyzer`.
+//!
+//! When using `cargo-remote`, you can configure your editor to perform the the typical
+//! "check-on-save" remotely as well. The configuration for VSCode is as follows:
+//!
+//! ```json
+//! {
+//! 	"rust-analyzer.cargo.buildScripts.overrideCommand": [
+//! 		"cargo",
+//! 		"remote",
+//! 		"--build-env",
+//! 		"SKIP_WASM_BUILD=1",
+//! 		"--",
+//! 		"build",
+//! 		"--message-format=json",
+//! 		"--all-targets",
+//! 		"--all-features",
+//! 		"--tests",
+//! 		"--target-dir=target/rust-analyzer"
+//! 	],
+//! 	"rust-analyzer.check.overrideCommand": [
+//! 		"cargo",
+//! 		"remote",
+//! 		"--build-env",
+//! 		"SKIP_WASM_BUILD=1",
+//! 		"--",
+//! 		"check",
+//! 		"--workspace",
+//! 		"--message-format=json",
+//! 		"--all-targets",
+//! 		"--all-features",
+//! 		"--tests",
+//! 		"--target-dir=target/rust-analyzer"
+//! 	],
+//! }
+//! ```
```

Comment:
Since above we also provide the Lua, maybe it's worth doing the same here. WDYT?

```suggestion
//! ```

and the same in Lua for `neovim/nvim-lspconfig`:

//!
//! ```lua
//! ["rust-analyzer"] = {
//!   cargo = {
//!     buildScripts = {
//!       overrideCommand = {
//!         "cargo",
//!         "remote",
//!         "--build-env",
//!         "SKIP_WASM_BUILD=1",
//!         "--",
//!         "build",
//!         "--message-format=json",
//!         "--all-targets",
//!         "--all-features",
//!         "--tests",
//!         "--target-dir=target/rust-analyzer"
//!       }
//!     },
//!     check = {
//!       overrideCommand = {
//!         "cargo",
//!         "remote",
//!         "--build-env",
//!         "SKIP_WASM_BUILD=1",
//!         "--",
//!         "check",
//!         "--workspace",
//!         "--message-format=json",
//!         "--all-targets",
//!         "--all-features",
//!         "--tests",
//!         "--target-dir=target/rust-analyzer"
//!       },
//!     },
//!   },
//! },
//! ```
```

Disclaimer: I haven't checked this but this is a lua translation. @liamaharon please confirm.

Reply:
Done. I never even considered that the build task should really just run check. Thanks for all the extra minutes of my life that I will now have 😂